jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/listener/ELContextCleaner.java

```diff
@@ -34,7 +33,9 @@
  */
 public class ELContextCleaner implements ServletContextListener
 {
-    private static final Logger LOG = LoggerFactory.getLogger(ELContextCleaner.class);
+    // IMPORTANT: This class cannot have a slf4j Logger
```

Comment:
In Jetty 10 and Jetty 11's jetty-maven-plugin IT tests there is an slf4j-impl provided by the project's test scope `jetty-slf4j-impl`.
In Jetty 10 and Jetty 11's test-distribution tests that use ELContextClassloader there is also a slf4j implementation present.

BUT ...

If you use jetty-maven-plugin from Jetty 10 or Jetty 11 or Jetty 12 on a project as it normally is, without declaring an slf4j implementation.
Then the webapp cannot load and causes a 503 unavailable error because the slf4j implementation wasn't found. (ClassNotFoundError)

Reply:
So if there's no slf4j impl then none of the jetty classes would work properly as they all depend on the slf4j api, so nothing special about the `ELContextCleaner` or any of the other listener classes that are exposed from `org.eclipse.jetty.servlet.listener`.  

BTW normally if slf4j can't find a logger impl it instantiates the NoOpLogger impl, so that should be the case with the plugin too?

Here's some output from running the jetty-10 plugin in `FORKED` mode where the forked process has no slf4j impl present. As you see, the NoOpLogger is used:

> [INFO] --- jetty-maven-plugin:10.0.12-SNAPSHOT:run (default-cli) @ test-new-maven-plugin ---
> [INFO] Configuring Jetty for project: Test :: Jetty Test New Maven Plugin Webapp
> [INFO] Classes = /home/janb/src/tmp/test-maven-plugin-refactor/target/classes
> [INFO] Context path = /test
> [INFO] Tmp directory = /home/janb/src/tmp/test-maven-plugin-refactor/target/tmp
> [INFO] web.xml file = file:///home/janb/src/tmp/test-maven-plugin-refactor/src/main/webapp/WEB-INF/web.xml
> [INFO] Webapp directory = /home/janb/src/tmp/test-maven-plugin-refactor/src/main/webapp
> [INFO] Web defaults = org/eclipse/jetty/webapp/webdefault.xml
> [INFO] Web overrides =  none
> [INFO] Scan interval sec = 1
> [INFO] Generating file:///home/janb/src/tmp/test-maven-plugin-refactor/target/fork-web.xml
> [WARNING] The test-jetty webapp is deployed. DO NOT USE IN PRODUCTION!
> [INFO] Generated file:///home/janb/src/tmp/test-maven-plugin-refactor/target/fork-web.xml
> [INFO] Started o.e.j.m.p.MavenWebAppContext@4777f71e{Test WebApp,/test,[file:///home/janb/src/tmp/test-maven-plugin-refactor/src/main/webapp/],AVAILABLE}{file:///home/janb/src/tmp/test-maven-plugin-refactor/src/main/webapp/}
> [INFO] Stopped o.e.j.m.p.MavenWebAppContext@4777f71e{Test WebApp,/test,[file:///home/janb/src/tmp/test-maven-plugin-refactor/src/main/webapp/],STOPPED}{file:///home/janb/src/tmp/test-maven-plugin-refactor/src/main/webapp/}
> [INFO] Forked process starting
> SLF4J: No SLF4J providers were found.
> SLF4J: Defaulting to no-operation (NOP) logger implementation
> SLF4J: See http://www.slf4j.org/codes.html#noProviders for further details.

Reply:
Correct, but now try to add an slf4j provider.
Each mode in jetty-maven-plugin has a different technique (embedded vs forked).
You'll quickly see that it's not easy.

Reply:
> @joakime looks like there is a clash with a recent change to the deployer (I removed a lot of the Resource usage and made it just use Path, as that is what the scanner is providing and it was silly to change to string then to resource then to File).
> 
> I'll ask Jan about the maven plugin stuff....

The conflict is resolved, been merged / fixed in this branch.

Reply:
slf4j state restored in 00e8921e042550599e34d50d9830245099cb391d

Reply:
Here's an example of what happens in Jetty-12 when the ELContextCleaner has sfl4j.

```
[INFO] jetty-12.0.0-SNAPSHOT; built: 2022-07-11T18:25:42.282Z; git: 67d03825080c75831e682a2a2b64665901d4f139; jvm 17.0.3+7
[WARNING] Failed startup of context o.e.j.e.m.p.MavenWebAppContext@31834a2b{ROOT,/,[file:///home/joakim/code/jetty/jetty.project-12.0.x/tests/jetty-ee9-test-maven-plugin/target/it/jetty-start-mojo-multi-module-single-war-it/webapp-war/src/main/webapp/],false}{file:///home/joakim/code/jetty/jetty.project-12.0.x/tests/jetty-ee9-test-maven-plugin/target/it/jetty-start-mojo-multi-module-single-war-it/webapp-war/src/main/webapp/}
java.lang.NoClassDefFoundError: org/slf4j/spi/SLF4JServiceProvider
    at java.lang.ClassLoader.defineClass1 (Native Method)
    at java.lang.ClassLoader.defineClass (ClassLoader.java:1012)
    at java.security.SecureClassLoader.defineClass (SecureClassLoader.java:150)
    at java.net.URLClassLoader.defineClass (URLClassLoader.java:524)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:427)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:421)
    at java.security.AccessController.doPrivileged (AccessController.java:712)
    at java.net.URLClassLoader.findClass (URLClassLoader.java:420)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClassFromSelf (ClassRealm.java:425)
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:42)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:271)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:247)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:239)
    at org.eclipse.jetty.ee9.webapp.WebAppClassLoader.loadClass (WebAppClassLoader.java:511)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:520)
    at java.lang.Class.forName0 (Native Method)
    at java.lang.Class.forName (Class.java:467)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass (ServiceLoader.java:1217)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService (ServiceLoader.java:1228)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext (ServiceLoader.java:1273)
    at java.util.ServiceLoader$2.hasNext (ServiceLoader.java:1309)
    at java.util.ServiceLoader$3.hasNext (ServiceLoader.java:1393)
    at org.slf4j.LoggerFactory.findServiceProviders (LoggerFactory.java:104)
    at org.slf4j.LoggerFactory.bind (LoggerFactory.java:147)
    at org.slf4j.LoggerFactory.performInitialization (LoggerFactory.java:139)
    at org.slf4j.LoggerFactory.getProvider (LoggerFactory.java:421)
    at org.slf4j.LoggerFactory.getILoggerFactory (LoggerFactory.java:407)
    at org.slf4j.LoggerFactory.getLogger (LoggerFactory.java:356)
    at org.slf4j.LoggerFactory.getLogger (LoggerFactory.java:382)
    at org.eclipse.jetty.ee9.servlet.listener.ELContextCleaner.<clinit> (ELContextCleaner.java:37)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstanceWithCaller (Constructor.java:499)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:480)
    at org.eclipse.jetty.ee9.nested.ContextHandler$APIContext.createInstance (ContextHandler.java:2184)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler$ServletAPIContext.createInstance (ServletContextHandler.java:1269)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler$ServletAPIContext.createInstance (ServletContextHandler.java:1278)
    at org.eclipse.jetty.ee9.servlet.BaseHolder.createInstance (BaseHolder.java:204)
    at org.eclipse.jetty.ee9.servlet.ListenerHolder.createInstance (ListenerHolder.java:100)
    at org.eclipse.jetty.ee9.servlet.ListenerHolder.doStart (ListenerHolder.java:89)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler.startContext (ServletContextHandler.java:370)
    at org.eclipse.jetty.ee9.webapp.WebAppContext.startContext (WebAppContext.java:1294)
    at org.eclipse.jetty.ee9.nested.ContextHandler.doStart (ContextHandler.java:644)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler.doStart (ServletContextHandler.java:294)
    at org.eclipse.jetty.ee9.webapp.WebAppContext.doStart (WebAppContext.java:516)
    at org.eclipse.jetty.ee9.maven.plugin.MavenWebAppContext.doStart (MavenWebAppContext.java:294)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:114)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.server.handler.ContextHandler.lambda$doStart$0 (ContextHandler.java:517)
    at org.eclipse.jetty.server.handler.ContextHandler$Context.call (ContextHandler.java:888)
    at org.eclipse.jetty.server.handler.ContextHandler.doStart (ContextHandler.java:517)
    at org.eclipse.jetty.ee9.nested.ContextHandler$CoreContextHandler.doStart (ContextHandler.java:2485)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:121)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:121)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.server.Server.start (Server.java:477)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:121)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.server.Server.doStart (Server.java:417)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.ee9.maven.plugin.JettyEmbedder.doStart (JettyEmbedder.java:225)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.ee9.maven.plugin.JettyStartMojo.startJettyEmbedded (JettyStartMojo.java:54)
    at org.eclipse.jetty.ee9.maven.plugin.AbstractWebAppMojo.startJetty (AbstractWebAppMojo.java:441)
    at org.eclipse.jetty.ee9.maven.plugin.AbstractWebAppMojo.execute (AbstractWebAppMojo.java:419)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.ClassNotFoundException: org.slf4j.spi.SLF4JServiceProvider
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:50)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:271)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:247)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:239)
    at java.lang.ClassLoader.defineClass1 (Native Method)
    at java.lang.ClassLoader.defineClass (ClassLoader.java:1012)
    at java.security.SecureClassLoader.defineClass (SecureClassLoader.java:150)
    at java.net.URLClassLoader.defineClass (URLClassLoader.java:524)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:427)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:421)
    at java.security.AccessController.doPrivileged (AccessController.java:712)
    at java.net.URLClassLoader.findClass (URLClassLoader.java:420)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClassFromSelf (ClassRealm.java:425)
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:42)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:271)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:247)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:239)
    at org.eclipse.jetty.ee9.webapp.WebAppClassLoader.loadClass (WebAppClassLoader.java:511)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:520)
    at java.lang.Class.forName0 (Native Method)
    at java.lang.Class.forName (Class.java:467)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass (ServiceLoader.java:1217)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService (ServiceLoader.java:1228)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext (ServiceLoader.java:1273)
    at java.util.ServiceLoader$2.hasNext (ServiceLoader.java:1309)
    at java.util.ServiceLoader$3.hasNext (ServiceLoader.java:1393)
    at org.slf4j.LoggerFactory.findServiceProviders (LoggerFactory.java:104)
    at org.slf4j.LoggerFactory.bind (LoggerFactory.java:147)
    at org.slf4j.LoggerFactory.performInitialization (LoggerFactory.java:139)
    at org.slf4j.LoggerFactory.getProvider (LoggerFactory.java:421)
    at org.slf4j.LoggerFactory.getILoggerFactory (LoggerFactory.java:407)
    at org.slf4j.LoggerFactory.getLogger (LoggerFactory.java:356)
    at org.slf4j.LoggerFactory.getLogger (LoggerFactory.java:382)
    at org.eclipse.jetty.ee9.servlet.listener.ELContextCleaner.<clinit> (ELContextCleaner.java:37)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstanceWithCaller (Constructor.java:499)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:480)
    at org.eclipse.jetty.ee9.nested.ContextHandler$APIContext.createInstance (ContextHandler.java:2184)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler$ServletAPIContext.createInstance (ServletContextHandler.java:1269)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler$ServletAPIContext.createInstance (ServletContextHandler.java:1278)
    at org.eclipse.jetty.ee9.servlet.BaseHolder.createInstance (BaseHolder.java:204)
    at org.eclipse.jetty.ee9.servlet.ListenerHolder.createInstance (ListenerHolder.java:100)
    at org.eclipse.jetty.ee9.servlet.ListenerHolder.doStart (ListenerHolder.java:89)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler.startContext (ServletContextHandler.java:370)
    at org.eclipse.jetty.ee9.webapp.WebAppContext.startContext (WebAppContext.java:1294)
    at org.eclipse.jetty.ee9.nested.ContextHandler.doStart (ContextHandler.java:644)
    at org.eclipse.jetty.ee9.servlet.ServletContextHandler.doStart (ServletContextHandler.java:294)
    at org.eclipse.jetty.ee9.webapp.WebAppContext.doStart (WebAppContext.java:516)
    at org.eclipse.jetty.ee9.maven.plugin.MavenWebAppContext.doStart (MavenWebAppContext.java:294)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:114)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.server.handler.ContextHandler.lambda$doStart$0 (ContextHandler.java:517)
    at org.eclipse.jetty.server.handler.ContextHandler$Context.call (ContextHandler.java:888)
    at org.eclipse.jetty.server.handler.ContextHandler.doStart (ContextHandler.java:517)
    at org.eclipse.jetty.ee9.nested.ContextHandler$CoreContextHandler.doStart (ContextHandler.java:2485)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:121)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:121)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.start (ContainerLifeCycle.java:171)
    at org.eclipse.jetty.server.Server.start (Server.java:477)
    at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart (ContainerLifeCycle.java:121)
    at org.eclipse.jetty.server.Handler$Abstract.doStart (Handler.java:391)
    at org.eclipse.jetty.server.Server.doStart (Server.java:417)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.ee9.maven.plugin.JettyEmbedder.doStart (JettyEmbedder.java:225)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start (AbstractLifeCycle.java:93)
    at org.eclipse.jetty.ee9.maven.plugin.JettyStartMojo.startJettyEmbedded (JettyStartMojo.java:54)
    at org.eclipse.jetty.ee9.maven.plugin.AbstractWebAppMojo.startJetty (AbstractWebAppMojo.java:441)
    at org.eclipse.jetty.ee9.maven.plugin.AbstractWebAppMojo.execute (AbstractWebAppMojo.java:419)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] Started oejen.ContextHandler$CoreContextHandler@40bf4386{ROOT,/,[file:///home/joakim/code/jetty/jetty.project-12.0.x/tests/jetty-ee9-test-maven-plugin/target/it/jetty-start-mojo-multi-module-single-war-it/webapp-war/src/main/webapp/],false}
[INFO] Started ServerConnector@2f677247{HTTP/1.1, (http/1.1)}{0.0.0.0:41631}
[INFO] Started oejs.Server@31acfd4e{STARTING}[12.0.0-SNAPSHOT,sto=0] @2268ms
```

This is because we are forcing the webapps to have slf4j present, even just the `slf4j-api` jar.

Reply:
The resources changes broken the jetty-ee#-maven-plugin testing.
I had to make changes there to get a green build.

Reply:
Opened Issue #8281 for a review of these changes.